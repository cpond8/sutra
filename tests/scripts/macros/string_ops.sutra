;; Macro string operations test
;; Tests macros that work with string values (no concatenation)

;; Macro that takes a string parameter
(define (echo_string text) (print text))
(echo_string "Hello")
(echo_string "World")

;; Macro that prints predefined strings
(define (say_hello name) (print name))
(say_hello "Alice")
(say_hello "Bob")

;; Macro with multiple string parameters
(define (print_two first second)
  (do
    (print first)
    (print second)))
(print_two "Line 1" "Line 2")
