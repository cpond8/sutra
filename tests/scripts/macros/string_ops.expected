Hello
World
Alice
Bob
Line 1
Line 2
foobar
abc
hello, world!
